utils/wasm-builder/src/lib.rs

```diff
@@ -248,20 +266,40 @@ impl CargoCommand {
 		cmd
 	}
 
-	/// Check if the supplied cargo command is a nightly version
-	fn is_nightly(&self) -> bool {
+	fn extract_version(program: &str, args: &[&str]) -> Option<Version> {
+		let version = Command::new(program)
+			.args(args)
+			.arg("--version")
+			.output()
+			.ok()
+			.and_then(|o| String::from_utf8(o.stdout).ok())?;
+
+		Version::extract(&version)
+	}
+
+	/// Returns the version of this cargo command or `None` if it failed to extract the version.
+	fn version(&self) -> Option<Version> {
+		self.version
+	}
+
+	/// Check if the supplied cargo command supports our Substrate wasm environment.
+	///
+	/// This means that either the cargo version is at minimum 1.68.0 or this is a nightly cargo.
+	///
+	/// Assumes that cargo version matches the rustc version.
+	fn supports_substrate_wasm_env(&self) -> bool {
 		// `RUSTC_BOOTSTRAP` tells a stable compiler to behave like a nightly. So, when this env
 		// variable is set, we can assume that whatever rust compiler we have, it is a nightly
 		// compiler. For "more" information, see:
 		// https://github.com/rust-lang/rust/blob/fa0f7d0080d8e7e9eb20aa9cbf8013f96c81287f/src/libsyntax/feature_gate/check.rs#L891
-		env::var("RUSTC_BOOTSTRAP").is_ok() ||
-			self.command()
-				.arg("--version")
-				.output()
-				.map_err(|_| ())
-				.and_then(|o| String::from_utf8(o.stdout).map_err(|_| ()))
-				.unwrap_or_default()
-				.contains("-nightly")
+		if env::var("RUSTC_BOOTSTRAP").is_ok() {
+			return true
+		}
+
+		let Some(version) = self.version() else { return false };
+
+		// Check if major and minor are greater or equal than 1.68 or this is a nightly.
+		version.major >= 1 && version.minor >= 68 || version.is_nightly
```

Comment:
```suggestion
		version.major > 1 || (version.major == 1 && version.minor >= 68) || version.is_nightly
```

We might never actually get version 2 so it probably doesn't matter, but just in case?